### DISCUSSION TITLE
Fork and maintain RulerZ

### DISCUSSION THREAD

**Discussion by yguedidi** [wallabag/wallabag#6802] [Ideas]
Hi!

Working on the upgrade to Symfony 5, one blockage is RulerZ dependencies.
When running `composer why-not symfony/symfony 5.4.27` on master there is this:

```            
kphoen/rulerz        0.21.1     requires         symfony/property-access (~3.0|~4.0)
kphoen/rulerz-bridge 1.1.1      requires         symfony/form (^3.0|^4.0)
kphoen/rulerz-bridge 1.1.1      requires         symfony/validator (^3.0|^4.0)
kphoen/rulerz-bundle 0.15.0     requires         symfony/framework-bundle (^3.0|^4.0)
```

See that Symfony 5+ are not allowed.
I suggest to fork those 3 repositories under the wallabag organization, and update them so that they support higher version of Symfony if possible. I can give it a try :)

Of course, scope is only about maintenance first, no expectation to bring new features or more.

What do you think?

At least this is a good first step, because better change would be to migrate to another solution, maybe Symfony Expression Language, but the work there will be much more complex!

**Comment by yguedidi** [comment]
Looking at usages of RulerZ it may not be complicated to use ExpressionLanguage instead.
What will be hard is the migration of data from RulerZ format to EP format 😅

**Reply by Kdecherf** [reply]
Would you be able to provide some examples of ExpressionLanguage/EP?

**Reply by yguedidi** [reply]
Sure! Already drafted something to attempt an upgrade to Symfony 5 on my local haha!

Here what I found:
- way more supported operators in EL
- no way to restrict to a subset of operators like possible with RulerZ
- possible to restrict possible variables in EL like RulerZ
- no equivalent of `matches` and `notmatches` RulerZ operators, and no possibility to add our operators
- we can add our custom functions to EL, so maybe something like `contains()` and `notcontains()`
- moving from an operator to a function means that the structure of a rule changes, from `foo matches "bar"` to `contains(foo, "bar")` I think (maybe need more investiagtion)

Based on all this, I think it's preferable to keep RulerZ for now, so forking and maintaining the repositories for upgrade only.

**Comment by yguedidi** [comment]
@Kdecherf do you think you can fork those repositories in the wallabag organization so I can start working on making tem support Symfony 5? :)

**Reply by nicosomb** [reply]
https://github.com/wallabag/rulerz
https://github.com/wallabag/RulerZBundle
https://github.com/wallabag/rulerz-bridge

**Reply by yguedidi** [reply]
Thank you @nicosomb! I'll start the work as soon as I can

**Reply by yguedidi** [reply]
@nicosomb looks like I'm not admin of those repositories 😅

**Reply by nicosomb** [reply]
@yguedidi Fixed.

**Comment by j0k3r** [comment]
Good idea @yguedidi
